pipelines/winget-cli-restsource-ci-pr.yml

```diff
@@ -8,51 +8,68 @@ variables:
-  - template: templates/copy-and-publish.yml
-    parameters:
-      name: StaticAnalysisLogs
-      source: '$(Agent.BuildDirectory)\_sdt\logs'
-
-  # SDT Post Analysis
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@1
-    displayName: 'Post Analysis'
-    inputs:
-      CredScan: true
+resources:
+  repositories:
+  - repository: 1ESPipelineTemplates
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+
+extends:
+  template: v1/1ES.Unofficial.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    pool:
+      name: Azure-Pipelines-1ESPT-ExDShared
+      image: windows-2022
+      os: windows
+    customBuildTags:
+    # This is added my 1ES migration tool and it is okay to remove in the future.
+    - ES365AIMigrationTooling
+
+    stages:
+    - stage: Build_Publish_Test_And_GuardianChecks
+      jobs:
+      - job: 'BuildPublishTest'
+        displayName: 'Build Publish & Tests'
+        timeoutInMinutes: 60
+        variables:
+        - name: BuildConfiguration
+          value: 'release'
+        - name: BuildPlatform
+          value: 'Any CPU'
+
+        steps:
+        # Restore and Build
+        - template: pipelines/templates/restore-build-publish-test.yml@self
+
+        templateContext:
+          outputs:
+          # Publish ARM Templates
+          - output: pipelineArtifact
+            displayName: 'Publish Artifact: WinGet.RestSource-WinGet.Restsource.Infrastructure'
+            targetPath: $(Build.ArtifactStagingDirectory)\WinGet.Restsource.Infrastructure
+            artifactName: WinGet.RestSource-WinGet.Restsource.Infrastructure
+
+          # Publish scripts
+          - output: pipelineArtifact
+            displayName: 'Publish Artifact: WinGet.RestSource-ReleaseScripts'
+            targetPath: $(Build.ArtifactStagingDirectory)\ReleaseScripts
+            artifactName: WinGet.RestSource-ReleaseScripts
+
+          # Publish Rest Function App
+          - output: pipelineArtifact
+            displayName: 'Publish Artifact: WinGet.RestSource-WinGet.RestSource.Functions'
+            targetPath: $(Build.ArtifactStagingDirectory)\WinGet.RestSource.Functions
+            artifactName: WinGet.RestSource-WinGet.RestSource.Functions
+
+          # Publish Rest Function App as an artifact 
+          - output: pipelineArtifact
+            displayName: 'Publish Artifact: WinGet.RestSource-WinGet.RestSource.IntegrationTest'
+            targetPath: $(Build.ArtifactStagingDirectory)\WinGet.RestSource.IntegrationTest
+            artifactName: WinGet.RestSource-WinGet.RestSource.IntegrationTest
+
+          # Publish Powershell Module
+          - output: pipelineArtifact
+            displayName: 'Publish Artifact: WinGet.RestSource-Winget.PowerShell.Source'
+            targetPath: '$(Build.ArtifactStagingDirectory)\Winget.PowerShell.Source'
+            artifactName: 'WinGet.RestSource-Winget.PowerShell.Source'
```

Comment:
I think we could have `$(Build.ArtifactStagingDirectory)` be the single output for the pipeline and not have all of the separate artifacts.

If we have stuff in that folder we don't want to publish, we could create something like `$(Build.ArtifactStagingDirectory)\publish` and put there only what we want to publish.

Of course, this is only if we don't have another pipeline or automated process that consumes  these artifacts. If it is only there so we can validate or we get them manually, having a single artifact is probably not a big deal.

Reply:
```suggestion
          - output: pipelineArtifact
            displayName: 'Publish Output Artifact'
            targetPath: $(Build.ArtifactStagingDirectory)
            artifactName: WinGet.RestSource
```

Reply:
Same comments in this file

Reply:
I will retain these for now because it will facilitate a one-to-one mapping between the old and new pipeline artifacts comparison. However, I agree that it is a good suggestion to combine them all as one build artifact.